src/views/ApplyPage/index.tsx

```diff
@@ -39,11 +31,18 @@ const ApplyPage = () => {
       <TextBox label="이메일" required formObject={formObject}>
         <InputLine
           label="이메일"
-          button={<CheckButton />}
           placeholder="이메일을 입력하세요"
           pattern={/^[0-9]*$/}
-          errorText="숫자만 입력해주세요"
-        />
+          errorText="숫자만 입력해주세요">
+          <InputButton
+            text="인증하기"
+            disabled={isActive}
+            onClick={() => {
+              setActive(true);
+            }}
+          />
+          <Timer isActive={isActive} setActive={setActive} />
```

Comment:
setActive가 Timer에서 setActive(false)로 밖에 안 쓰이더라고요
setActive 전체를 넘기기보다는 isActive를 false로 바꾸는 용으로 사용하도록 제한을 두는 게 좋을 거 같습니다!
```suggestion
          <Timer isActive={isActive} onResetTimer={() => { setActive(false); } />
```

그렇게 했을 때의 장점

1. 예상 밖의 일 일어날 확률 감소. 무조건 false로만 설정되므로.
2. handler를 prop으로 전달할 때는 보통 접두사 on을 붙이므로 컨벤션 통일할 수 있음.
3. type 정의할 때, 긴 값들 import 할 필요 없이 () => void 하면 끝남.
```tsx
setActive: React.Dispatch<React.SetStateAction<boolean>>;
// 대신 아래 처럼 가능
onResetTimer: () => void;
```

(+) 통상 setState 네이밍을 지을 땐 state 변수명을 따라가므로 setActive대신 setIsActive로 변경하면 좋을 거 같습니다!

Reply:
너무 좋은 것 같습니다